File: 240-Search-a-2D-Matrix-II.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int left=0;
        int right=matrix[0].size()-1;
        while(left<matrix.size() && right>=0){
            if(matrix[left][right]==target){
                return true;
            }
            else if(matrix[left][right]<target){
                left=left+1;
            }
            else{
                right=right-1;
            }
        }
        return false;
    }
};

File: 240-Search-a-2D-Matrix-II.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        left=0
        right=len(matrix[0])-1
        while left<len(matrix) and right>=0:
            if matrix[left][right]==target:
                return True
            elif matrix[left][right]<target:
                left=left+1
            else:
                right=right-1
        return False